### ISSUE TITLE
Under restricted psa, ephemeral_container can't be created with restricted profile

### ISSUE THREAD

**Issue by tonyaw** [kubernetes/kubernetes#118961]
### What happened?

I assume ephemeral container shall be able to create via restricted debugging profile under restricted PS.
Unfortunately, it is failed:
[root@foss-ssc-11 ~]# kubectl debug -ti --image=localhost/tonyaw/tester:v0.9 ephemeral-demo-b78475778-hmpl7 --target=ephemeral-demo --profile=restricted -n psa-test -- sh
Targeting container "ephemeral-demo". If you don't see processes from this container it may be because the container runtime doesn't support this feature.
Defaulting debug container name to debugger-qntwj.
Error from server (Forbidden): pods "ephemeral-demo-b78475778-hmpl7" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (containers "debugger", "debugger-2hf9x", "debugger-cg5sg", "debugger-gn9c6", "debugger-8lxd2", "debugger-tj95g", "debugger-b2wtr", "debugger-6zmr2", "debugger-7fkjw", "debugger-dpqwr", "debugger-qntwj" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (containers "debugger", "debugger-2hf9x", "debugger-cg5sg", "debugger-gn9c6", "debugger-8lxd2", "debugger-tj95g", "debugger-b2wtr", "debugger-6zmr2", "debugger-7fkjw" must set securityContext.capabilities.drop=["ALL"]; containers "debugger-tj95g", "debugger-6zmr2" must not include "NET_ADMIN", "SYS_PTRACE" in securityContext.capabilities.add), runAsNonRoot != true (pod or containers "debugger", "debugger-2hf9x", "debugger-cg5sg", "debugger-gn9c6", "debugger-8lxd2", "debugger-tj95g", "debugger-b2wtr", "debugger-6zmr2", "debugger-7fkjw" must set securityContext.runAsNonRoot=true), seccompProfile (pod or containers "debugger", "debugger-2hf9x", "debugger-cg5sg", "debugger-gn9c6", "debugger-8lxd2", "debugger-tj95g", "debugger-b2wtr", "debugger-6zmr2", "debugger-7fkjw", "debugger-dpqwr", "debugger-qntwj" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
[root@foss-ssc-11 ~]# 

### What did you expect to happen?

ephemeral container shall be able to create via restricted debugging profile under restricted PS.
If not, what is the propose of restricted debugging profile?

### How can we reproduce it (as minimally and precisely as possible)?

Under a restricted PS namespace:
kubectl debug -ti --image=localhost/tonyaw/tester:v0.9 ephemeral-demo-b78475778-hmpl7 --target=ephemeral-demo **--profile=restricted** -n psa-test -- sh

### Kubernetes version

<details>

```console
[root@foss-ssc-11 ~]# kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.3", GitCommit:"25b4e43193bcda6c7328a6d147b1fb73a33f1598", GitTreeState:"clean", BuildDate:"2023-06-14T09:53:42Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.3", GitCommit:"25b4e43193bcda6c7328a6d147b1fb73a33f1598", GitTreeState:"clean", BuildDate:"2023-06-14T09:47:40Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}
[root@foss-ssc-11 ~]#
```

</details>

### Cloud provider

<details>
Local server
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by mukeshmahato18** [comment]
This issue appears to be related to the creation of an ephemeral container using a restricted debugging profile within a restricted PodSecurity (PS) namespace. The user attempted to create the ephemeral container using the kubectl debug command with specific parameters, but encountered an error.

The error message indicates that the pod "ephemeral-demo-b78475778-hmpl7" violates the PodSecurity policy named "restricted:latest" due to several conditions not being met. The error message lists the specific violations:

    allowPrivilegeEscalation is not set to false for the containers.
    The containers have unrestricted capabilities and must set securityContext.capabilities.drop to ["ALL"].
    The containers include "NET_ADMIN" and "SYS_PTRACE" in securityContext.capabilities.add, which is not allowed.
    runAsNonRoot is not set to true for the pod or containers.
    seccompProfile is not set to "RuntimeDefault" or "Localhost" for the pod or containers.

These violations prevent the creation of the ephemeral container using the restricted debugging profile. The purpose of the restricted debugging profile is to provide a more secure environment for debugging, but in this case, the restrictions specified in the PodSecurity policy are not being met.

**Comment by tonyaw** [comment]
> 

Right.
So my first question is if the following use use case is valid:
I shall be able to create ephemeral container with restricted debugging profile under a restricted PS namespace successfully.

If not, could you please provide what use case shall be for restricted debugging profile?

**Comment by SergeyKanzhelev** [comment]
/sig cli

Seems to be similar to https://github.com/kubernetes/kubernetes/issues/118962, mentioning the KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/1441-kubectl-debug#profile-netadmin

**Comment by mochizuki875** [comment]
@tonyaw 
Hi, it seems to have been fixed and merged.
Planed to release in v1.28.

#117405
#117543

Thanks.

**Comment by tonyaw** [comment]
> @tonyaw Hi, it seems to have been fixed and merged. Planed to release in v1.28.
> 
> #117405 #117543
> 
> Thanks.

Got it. Thanks!
Will test it after 1.28 is released. :-)

**Comment by mochizuki875** [comment]
@tonyaw 

FYI:
Now, you can test it before released v1.28.
(It's only for testing)

Download and install go v1.20.x.
https://go.dev/doc/install

Clone k/k repository.
```console
$ git clone https://github.com/kubernetes/kubernetes.git
```

Build `kubectl`.
```
$ cd kubernetes/cmd/kubectl
$ go build ./kubectl.go
```

Now, you can get `kubectl` binary and test it.
```
$ ls
kubectl  kubectl.go  OWNERS
$ ./kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-master+$Format:%H$", GitCommit:"$Format:%H$", GitTreeState:"", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"go1.20", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.2", GitCommit:"7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647", GitTreeState:"clean", BuildDate:"2023-05-17T14:13:28Z", GoVersion:"go1.20.4", Compiler:"gc", Platform:"linux/amd64"}
error: could not parse pre-release/metadata (-master+$Format:%H$) in version "v0.0.0-master+$Format:%H$"
```

**Comment by mpuckett159** [comment]
/triage accepted
As this was intended to be fixed in the 1.28 release, if you could verify the fix works for you (either after 1.28 is released or by using the master branch build) and please close the issue, otherwise the repo bot will close it for us.

**Comment by tonyaw** [comment]
I will close this ticket for now. 
If I meet any issue during verification, I will consult you again.
Thanks for your great support very much! :-)
